[MICI] ui: better confidence ball

royjr opened the pull request:
remove gradient to better differenciate between engaged and lat only states. we can also use the tizi state colors instead of mici to keep it uniform across devices

**engaged (before)**
<img width="537" height="237" alt="Screenshot 2025-12-29 at 19 29 16" src="https://github.com/user-attachments/assets/fc070912-2f5d-47d7-b418-de44fe12f207" />

**engaged (after)**
<img width="535" height="236" alt="Screenshot 2025-12-29 at 19 28 05" src="https://github.com/user-attachments/assets/ed5bf6b7-2aa2-404f-8d34-68efbb422ff1" />

**lat only**
<img width="536" height="236" alt="Screenshot 2025-12-29 at 19 28 37" src="https://github.com/user-attachments/assets/e03cc43f-d293-437a-bd36-79356c3a6211" />

royjr committed:
use default colors

royjr committed:
use default bottom_dot_color

royjr committed:
Revert "use default bottom_dot_color"

This reverts commit 3300db1235a37d5a2caf1ee43ac8b367050592a8.

royjr committed:
Revert "use default colors"

This reverts commit 6f020a2f66ed471429a678daf22a4f179da33ae3.

royjr committed:
remove gradient

royjr committed:
Merge branch 'master' into ui-better-confidence-ball

royjr committed:
optional

royjr committed:
Merge branch 'master' into ui-better-confidence-ball

royjr committed:
bool

devtekve committed:
Merge branch 'master' into ui-better-confidence-ball

devtekve commented on sunnypilot/sunnylink/params_metadata.json:
```diff
@@ -300,6 +300,10 @@
   },
+  "FlatUIState": {
+    "title": "Flat UI State",
+    "description": ""
```
Im not sure this would explain to anyone on sunnylink what this really does tho

devtekve reviewed (changes requested):
Just need to improve the description (and the title ?) so that someone with zero context can understand what is expected from this

royjr committed:
Merge branch 'master' into ui-better-confidence-ball

sunnyhaibin committed:
Merge branch 'master' into ui-better-confidence-ball

sunnyhaibin commented:
> Just need to improve the description (and the title ?) so that someone with zero context can understand what is expected from this

@royjr let me know when this is updated and ready for review again!

royjr committed:
Merge branch 'master' into ui-better-confidence-ball

royjr committed:
Rename FlatUIState to FlatConfidenceBall

Rename the parameter key FlatUIState to FlatConfidenceBall and update all references. Changes update common/params_keys.h, the UI state property (self.flat_confidence_ball in selfdrive/ui/sunnypilot/ui_state.py), the confidence ball rendering check (selfdrive/ui/mici/onroad/confidence_ball.py), and the params metadata (sunnypilot/sunnylink/params_metadata.json) with a clearer title/description

royjr committed:
Merge branch 'master' into ui-better-confidence-ball

royjr committed:
Merge branch 'master' into ui-better-confidence-ball

royjr committed:
Merge branch 'master' into ui-better-confidence-ball

royjr committed:
Merge branch 'master' into ui-better-confidence-ball
